src/DynamoCoreWpf/Views/Preview/WatchTree.xaml.cs

```diff
@@ -39,18 +40,53 @@ private void Button_Click(object sender, RoutedEventArgs e)
                 node.Click();
         }
 
-        private void TreeView1_OnSelectedItemChanged(object sender, RoutedPropertyChangedEventArgs<object> e)
+        private void treeviewItem_MouseLeftButtonUp(object sender, System.Windows.Input.MouseButtonEventArgs e)
         {
-            var node = e.NewValue as WatchViewModel;
-            if (node == null)
+            TreeViewItem tvi = (TreeViewItem)sender;
+            var node = tvi.DataContext as WatchViewModel;
+
+            HandleItemChanged(tvi, node);
+
+            e.Handled = true; 
+        }
+        private void treeviewItem_KeyUp(object sender, System.Windows.Input.KeyEventArgs e)
+        {
+            if (e.Key == System.Windows.Input.Key.Up || e.Key == System.Windows.Input.Key.Down)
+            {
+                TreeViewItem tvi = sender as TreeViewItem;
+                var node = tvi.DataContext as WatchViewModel;
+
+                HandleItemChanged(tvi, node);
+            }
+            e.Handled = true;
+        }
+
+        private void HandleItemChanged (TreeViewItem tvi, WatchViewModel node)
+        {
+            if (tvi == null || node == null)
                 return;
 
+            // checks to see if the currently selected node is the same as the previous selected node
+            // if so, then de-select the currently selected node.
+
+            if (prevVM == null || node.Path != this.prevVM.Path)
```

Comment:
No, there shouldn't be the case where node === prevVM but node.Path != prevVM.Path as the path of the node is unique to that particular node. But we should be able to simplify to `if (node == prevVM)`. 

prevVM is solely used in this comparison to check if the node currently being selected is the same as the selected node.

I'll also fix the name here of prevVM to prevWatchViewModel.